charts/spire/charts/spiffe-oidc-discovery-provider/templates/hpa.yaml

```diff
@@ -14,16 +14,20 @@ spec:
   minReplicas: {{ .Values.autoscaling.minReplicas }}
   maxReplicas: {{ .Values.autoscaling.maxReplicas }}
   metrics:
-    {{- if .Values.autoscaling.targetCPUUtilizationPercentage }}
+    {{- if .Values.autoscaling.targetMemoryUtilizationPercentage }}
     - type: Resource
       resource:
-        name: cpu
-        targetAverageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
+        name: memory
+        target:
+          type: Utilization
+          averageUtilization: {{ .Values.autoscaling.targetMemoryUtilizationPercentage }}
     {{- end }}
-    {{- if .Values.autoscaling.targetMemoryUtilizationPercentage }}
+    {{- if .Values.autoscaling.targetCPUUtilizationPercentage }}
     - type: Resource
       resource:
-        name: memory
-        targetAverageUtilization: {{ .Values.autoscaling.targetMemoryUtilizationPercentage }}
+        name: cpu
+        target:
+          type: Utilization
+          averageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
```

Comment:
Any specific reason to switch the order here? 

Can we leave the original order to reduce the diff in this PR?

Reply:
With the existing order, Argo-CD thinks that there is a diff because the order ends up being modified. The Argo-CD issue I mentioned in the PR suggests reordering the metrics as a possible workaround (in my case I do not own the argo configuration): https://github.com/argoproj/argo-cd/issues/1079#issuecomment-463849329